app/components/Auth/Form.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import {
+  StyleSheet,
+  Text,
+  TextInput,
+  View,
+} from 'react-native';
+import Button from 'react-native-button';
+
+class Form extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      email: '',
+      password: '',
+    }
+  }
+
+  updateEmail = email => this.setState({ email })
+  updatePassword = password => this.setState({ password })
+
+  onSubmit = () => {
+    const { email, password } = this.state
+    const { onSubmit } = this.props
+
+    onSubmit({ email, password })
+  }
+
+  render() {
+    const { authPage } = this.props;
+
+    return (
+      <View style={styles.container}>
+        <Text style={styles.pageText}>{authPage}</Text>
+
+        <TextInput
+          placeholder='Email'
+          autoCapitalize='none'
+          autoCorrect={false}
+          autoFocus={true}
+          style={styles.textInput}
+          keyboardType='email-address'
+          value={this.props.email}
+          onChangeText={this.updateEmail}
+        />
+        <TextInput
+          placeholder='Password'
+          autoCapitalize='none'
+          autoCorrect={false}
+          secureTextEntry={true}
+          style={styles.textInput}
+          value={this.props.password}
```

Comment:
destructure?

Reply:
How can I destructure this?

Reply:
`const {  password } = this.props` at the top of the function?